### PR TITLE
feat(ci): use self-hosted runner for faster main build

### PR DESCRIPTION
**Pull request by lidel**
The goal is to make it faster, this PR aims to test if using self-hosted runner speeds up release build.

If it is ok, we can decide to set CI_BUILD_RUNS_ON on this repo instead.

cc @galargh  for sanity check +  if we need to safelist this repo somewhere for this to work (i tried to set `CI_BUILD_RUNS_ON` but https://github.com/ipfs/distributions/actions/runs/5799417224/job/15719370344 was stuck waiting for free runner and i cancelled it).

### CODE REVIEW

**Comment by galargh** [comment]
> cc @galargh for sanity check + if we need to safelist this repo somewhere for this to work (i tried to set `CI_BUILD_RUNS_ON` but https://github.com/ipfs/distributions/actions/runs/5799417224/job/15719370344 was stuck waiting for free runner and i cancelled it).

Yes, we do! I'll do that first thing tomorrow morning.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ env:
 
 jobs:
   build:
-    runs-on: ${{ fromJSON(vars.CI_BUILD_RUNS_ON || '"ubuntu-latest"') }}
+    runs-on: ${{ fromJSON(vars.CI_BUILD_RUNS_ON) || fromJSON(github.repository == 'ipfs/distributions' && '["self-hosted", "linux", "x64", "2xlarge"]') || '"ubuntu-latest"') }}
```
Comment (galargh) [inline .github/workflows/main.yml:24]:
```suggestion
    runs-on: ${{ fromJSON(vars.CI_BUILD_RUNS_ON || '"ubuntu-latest"') }}
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -95,7 +95,7 @@ jobs:
         continue-on-error: true # skip if no releases
 
   persist:
-    runs-on: "ubuntu-latest"
+    runs-on: ${{ fromJSON(vars.CI_BUILD_RUNS_ON) || fromJSON(github.repository == 'ipfs/distributions' && '["self-hosted", "linux", "x64", "2xlarge"]') || '"ubuntu-latest"') }}
```
Comment (galargh) [inline .github/workflows/main.yml:98]:
```suggestion
    runs-on: ${{ fromJSON(vars.CI_BUILD_RUNS_ON || '"ubuntu-latest"') }}
```

**Comment by galargh** [comment]
We have access to self-hosted runners here now 🥳 but we do need to make some adjustments before we update the `CI_BUILD_RUNS_ON` var. E.g. https://github.com/ipfs/distributions/actions/runs/5836552866/job/15830374575 - we'll need to update the ipfs setup action so that it works on self-hosted. I'll get back to it next week.

**Comment by lidel** [comment]
@galargh  rebased, is the issue gone?

**Comment by galargh** [comment]
Seems so, I think we're good to go 😁

**Review by galargh — COMMENTED** [review]
We can set `CI_BUILD_RUNS_ON` var to `["self-hosted", "linux", "x64", "2xlarge"]` again.

After we merge it, that var will also affect the `persist` job.

**Pull request merged**